lms/djangoapps/courseware/views/views.py

```diff
@@ -863,6 +864,126 @@ def post(self, request, course_id):
         return redirect(reverse('about_course', args=[text_type(course_key)]))
 
 
+@ensure_csrf_cookie
+@cache_if_anonymous()
+def course_page(request, course_id):
+    discovery_course = get_course_from_discovery(course_id)
+    if not discovery_course:
+        raise Http404(u"Course key not found on discovery")
+
+    [course] = discovery_course
```

Comment:
a single object will work here instead of having a list?

Reply:
fixed